### PR TITLE
🌱 Allows configuring crossZoneLoadBalancing on controlPlaneLoadBalancer

### LINKED ISSUE #1754: AWSClusterSpec's "ControlPlaneLoadBalancer" field should be mutable
/kind bug

**What steps did you take and what happened:**

Create a CAPA cluster using the default settings. Observe that the API servers are fronted by a Classic ELB using the "Internet-facing" scheme, with cross-zone load balancing disabled.

Edit the Kubernetes manifest for the _AWSCluster_ object, adding a "controlPlaneLoadBalancer" field (be it in JSON or YAML), with its "crossZoneLoadBalancing" field set to `true`.

Apply the changes to the API server, and observe that it rejects the attempt, complaining that the "spec.controlPlaneLoadBalancer [...] field is immutable". Note, though, that no matter what I specify here—even for cases where it should match the current configuration once parsed—the API server rejects it. Here are some candidate updates:
- Specify matching values
```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSCluster
metadata:
  name: aws-test-1
  namespace: aws-test-1
spec:
  controlPlaneLoadBalancer:
    scheme: Internet-facing
    crossZoneLoadBalancing: true
  # ...
```

- Eliminate one field
```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSCluster
metadata:
  name: aws-test-1
  namespace: aws-test-1
spec:
  controlPlaneLoadBalancer:
    scheme: Internet-facing
  # ...
```

- Eliminate the other field
```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSCluster
metadata:
  name: aws-test-1
  namespace: aws-test-1
spec:
  controlPlaneLoadBalancer:
    crossZoneLoadBalancing: true
  # ...
```

- Eliminate both fields
```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSCluster
metadata:
  name: aws-test-1
  namespace: aws-test-1
spec:
  controlPlaneLoadBalancer:
  # ...
```

Observe that the field doesn't appear to be set on the existing Kubernetes object:
```
% kubectl --namespace=aws-test-1 get awscluster aws-test-1 --output=jsonpath='{.spec.controlPlaneLoadBalancer}'
<no output>
```

This is the only mention of the "controlPlaneLoadBalancer" field that the API server accepts:
```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
kind: AWSCluster
metadata:
  name: aws-test-1
  namespace: aws-test-1
spec:
  controlPlaneLoadBalancer: null
  # ...
```

**What did you expect to happen:**

The CAPA controllers would reconfigure the load balancer to allow cross-zone load balancing. At minimum, I'd expect it to accept specified values that match the current values (after defaults are applied).

**Anything else you would like to add:**
PR #1677 introduced the validation that rejects all mutation of the `AWSCluster.Spec.ControlPlaneLoadBalancer` field. That may have been in the interest of safety, if there is not yet any way to reconcile these changes to the fields. @benmoss, do you recall this one?

As far as I can tell, it's not possible to change the scheme of a Classic ELB after it's created, but [it _is_ possible to change its cross-zone load balancing setting](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-disable-crosszone-lb.html) (via [the _aws_ tool](https://docs.aws.amazon.com/cli/latest/reference/elb/modify-load-balancer-attributes.html),  or [the API](https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_ModifyLoadBalancerAttributes.html) with [this attribute](https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_CrossZoneLoadBalancing.html)).

**Environment:**

- Cluster-api-provider-aws version:  
  v0.5.4
- Kubernetes version: (use `kubectl version`):  
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-18T02:59:13Z", GoVersion:"go1.14.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-30T20:19:45Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`):  
  This is a _kind_ cluster running on macOS.
```
NAME="Ubuntu"
VERSION="19.10 (Eoan Ermine)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 19.10"
VERSION_ID="19.10"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=eoan
UBUNTU_CODENAME=eoan
```

### PR DESCRIPTION
**Pull request by gab-satchi**
- removed omit empty on the field so it still shows up when set to false

Fixes #1754

### CODE REVIEW

File: api/v1alpha3/awscluster_webhook_test.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func TestAWSCluster_ValidateUpdate(t *testing.T) {
 			wantErr: true,
 		},
 		{
-			name: "controlPlaneLoadBalancer is immutable",
+			name: "controlPlaneLoadBalancer scheme is immutable",
```
Comment (randomvariable) [inline api/v1alpha3/awscluster_webhook_test.go:48]: Add a test for allow changing of the cross zone aspect?

**Comment by randomvariable** [comment]
Could also a test to loadbalancer_test.go that s.configureAttributes does indeed get called when the CrossZoneLoadBalancing attribute is changed. WDYT

**Comment by gab-satchi** [comment]
the configureAttributes is just a private method so testing that's called alone may not be valuable. Adding tests around ensuring the correct AWS calls are made might be better. I think AWSMachine controller tests do a decent job at that so could do something similar with cluster.

**Comment by randomvariable** [comment]
SGTM. Merging this in and treat #1804 separately.
/lgtm
/approve

**Pull request merged**